### ISSUE TITLE
python_backend always tries to chmod the triton_python_backend_stub

### ISSUE THREAD

**Issue by okdimok** [triton-inference-server/server#3608]
**Description**
To check, whether the change of permission in needed for the triton_python_backend_stub file, the python backend tries to run it without any arguments, and then checks the exit code. See [here](https://github.com/triton-inference-server/python_backend/blob/main/src/python.cc#L1207).
However, this run always results in the exit code of 1, due to the number of the arguments being smaller, then required by the stub. See [here](https://github.com/triton-inference-server/python_backend/blob/main/src/pb_stub.cc#L714).
It makes the python_backend always try to chmod the triton_python_backend_stub file, which results in an error, if the container fs is mounted read-only (which is typical for Singularity).

**Triton Information**
The issue is still present in the main branch.
It can be replicated with 21.10 NGC container

**To Reproduce**

Add any python model to the model repository. Then 

```
singularity pull tritonserver-21.10-py3.sif docker://nvcr.io/nvidia/tritonserver:21.10-py3
SINGULARITYENV_LD_PRELOAD=/plugins/libyolo_layer_static.so singularity run --nv --bind $(pwd)/plugins:/plugins --bind model_repository:/models tritonserver-21.10-py3.sif tritonserver --model-repository=/models --strict-model-config=false --log-error=true --log-warning=true --log-info=true
```

It results in 
```
E1124 11:26:53.294130 281066 model_repository_manager.cc:1186] failed to load 'python_model' version 1: Internal: Failed to give execute permission to triton_python_backend_stub in /opt/tritonserver/backends/python/triton_python_backend_stub python_model_0 Error No.: -1
```

**Expected behavior**
No chmod needs to happen, if the triton_python_backend_stub already has the proper permissions. As a solution, one may suggest introducing a special arg set to triton_python_backend_stub, or just proper reading of the file permissions [here](https://github.com/triton-inference-server/python_backend/blob/main/src/python.cc#L1207).

**Comment by Tabrizian** [comment]
@okdimok Python backend doesn't always tries to chmod the stub. It only tries to chmod the stub when it doesn't have the **execute** permission. The `triton_python_backend_stub` needs to have execute and read permissions. Can you try running `/opt/tritonserver/backends/python/triton_python_backend_stub` without any arguments in your container and share the return code?

**Comment by tyommik** [comment]
@Tabrizian 
```
(pytorch38) ➜  triton ./start-triton-singularity.sh
# ls la /opt/tritonserver/backends/python/triton_python_backend_stub
-rwxr-xr-x 1 root root 489040 Oct 21 20:36 /opt/tritonserver/backends/python/triton_python_backend_stub
1124 11:16:17.836518 273635 pb_stub.cc:713] Expected 9 arguments, found 1 arguments.
# return code
1
# ls la /opt/tritonserver/backends/python/triton_python_backend_stub
-rwxr-xr-x 1 root root 489040 Oct 21 20:36 /opt/tritonserver/backends/python/triton_python_backend_stub
```

**Comment by Tabrizian** [comment]
Thanks for pointing out this issue. I have created a PR to fix this.

**Comment by dchebakov** [comment]
In which version this fix was introduced?

**Comment by Tabrizian** [comment]
The fix was added in 21.12 version of Triton.
